src/modules/users-permissions/PermissionManager.js

```diff
@@ -0,0 +1,35 @@
+
+class PermissionManager {
+  constructor(user, permissionObjs) {
+    this._user = user
+    this._permisionObjs = permissionObjs
+    
+    // construct user permissions list
+    this._permissions = new Set()
+    this._user.roles.forEach((role) => {
+      for (let p of role.permissions) {
+        this._permissions.add(p)
+      }
+    })
+  }
+
+  allow(permissionName, obj = null) {
```

Comment:
This has a different intent than the `UsersPermissions.allow()` correct?  One is a register function, the other is a check?  If so, I would make consistent or rename to clarify the different functionality.